examples/src/lib.rs

```diff
@@ -229,17 +230,30 @@ impl ExampleBase {
                 .to_vec();
             extension_names.push(DebugUtils::name().as_ptr());
 
+            #[cfg(any(target_os = "macos", target_os = "ios"))]
+            {
+                extension_names.push(KhrGetPhysicalDeviceProperties2Fn::name().as_ptr());
```

Comment:
This is a device extension and is required for `VK_KHR_portability_subset` below, not the `VK_KHR_portability_enumeration` **instance** extension here. Can you move it down and also add a comment that, while we're not using this extension directly, states it is a requirement for `KhrPortabilitySubsetFn`?

Reply:
Mmh, maybe I'm misunderstanding something here regarding the difference between instance and device extension, could you clarify for me? If I try to use it as a device extension, then the application crashes with

> ERROR:
> GENERAL [Loader Message (0)] : loader_validate_device_extensions: Device extension VK_KHR_get_physical_device_properties2 not supported by selected physical device or enabled layers.

[This doc](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VK_KHR_get_physical_device_properties2.html) also lists as an instance extension.

Reply:
It indeed is a **physical** device extension which implies it is an instance extension and not a **device** extension, I don't know what I was doing while writing this comment...